torch/csrc/jit/codegen/cuda/executor.cpp

```diff
@@ -957,6 +959,7 @@ KernelArgumentHolder FusionExecutor::inferOutputSizes(
 std::vector<at::Tensor> FusionExecutor::runFusion(
     KernelArgumentHolder& args,
     const LaunchParams& launch_constraints,
+    const int maxrregcount,
```

Comment:
Since we are giving `maxrregcount` here,  I think we should recompile the kernel if a different value is given with the one used for the compiled binary.

Reply:
added `maxrregcount_high_water_mark`, It is initilized to 255 and will be set to maxrregcount after first compile. Then, if requested `maxrregcount > maxrregcount_high_water_mark` , will recompile.